frontend/src/views/producer/dashboard/edition-form/CoverImage.jsx

```diff
@@ -20,15 +21,47 @@ const FlexBox = styled.div`
   width: 100%;
   align-items: center;
 `;
-
+const ErrorMessage = styled('span')`
+  p {
+    display: flex;
+    font-size: 1rem;
+    margin: 0;
+    text-align: left;
+    color: red;
+    display: ${props => (props.show ? 'flex' : 'none')};
+  }
+`;
 class CoverImage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.uploadImage = this.uploadImage.bind(this);
+  }
+
+  async uploadImage(ev) {
+    ev.preventDefault();
+    this.props.uploadProgress(true);
+    const imageurl = await this.props.store.uploadImage(ev.target.files[0]);
+    this.props.updateCoverImage(imageurl);
+  }
+
   render() {
-    const { inputBackgroundColor, value, onChange } = this.props;
+    const {
+      inputBackgroundColor,
+      value,
+      onChange,
+      updateCoverImage,
+      uploadProgress,
+      uploadFailed,
+    } = this.props;
     return (
       <Wrapper>
         <Typography style={{ transform: 'translateY(-1rem)' }} type="largebody">
           Cover image
+          <InputField type="file" onChange={this.uploadImage} />
```

Comment:
Now it instantly uploads when the file is selected. Don't you think it would be nicer for the user if there would be a button to initiate the upload after selecting the image?

Reply:
I specified the design that way though